src/compiler/checker.ts

```diff
@@ -18272,20 +18273,27 @@ namespace ts {
                             return Ternary.Maybe;
                         }
                     }
-                    if (depth === 100) {
+                    if (sourceDepth === 100 || targetDepth === 100) {
                         overflow = true;
                         return Ternary.False;
                     }
                 }
+                const constantSource = sourceStack[sourceDepth] === source;
+                const constantTarget = targetStack[targetDepth] === target;

```

Comment:
This doesn't seem right. Only elements at indices less than `sourceDepth` and `targetDepth` have meaningful values. This appears to compare an effectively random value with `source` or `target`. Or am I missing something?

Reply:
Don't you need something like this?

```ts
const constantSource = sourceDepth > 0 && sourceStack[sourceDepth - 1] === source;
const constantTarget = targetDepth > 0 && targetStack[targetDepth - 1] === target;
```